kuma/static/js/newsletter.js

```diff
@@ -0,0 +1,221 @@
+                    }
+                    var response = r.target.response;
+                    if (response.success === true) {
+                        newsletterForm.style.display = 'none';
+                        newsletterThanks();
+                    }
+                    else {
+                        if(response.errors) {
+                            for (var i = 0; i < response.errors.length; i++) {
+                                errorArray.push(response.errors[i]);
+                            }
+                        }
+                        newsletterError(new Error());
+                    }
+                }
+                else {
+                    newsletterError(new Error());
+                }
+            };
+
+            xhr.onerror = function(event) {
+                newsletterError(event);
+            };
+
+            var url = newsletterForm.getAttribute('action');
+
+            xhr.open('POST', url, true);
+            xhr.setRequestHeader('Content-type', 'application/x-www-form-urlencoded');
+            xhr.setRequestHeader('X-Requested-With','XMLHttpRequest');
+            xhr.timeout = 5000;
+            xhr.ontimeout = newsletterError;
+            xhr.responseType = 'json';
+            xhr.send(params);
+
+            return false;
+        }
+
+        newsletterForm.addEventListener('submit', newsletterSubscribe, false);
+
+        newsletterEmailInput.addEventListener('focus', function() {
+            newsletterPrivacy.style.display ='block';
+            mdn.analytics.trackEvent({
+                'category': 'newsletter',
+                'action': 'prompt',
+                'label': 'focus'
+            });
+        }, false);
+    }
+
+    function newsletterAddHideButton() {
+        // fallback text for button
+        var textSpan = doc.createElement('span');
+        textSpan.appendChild(doc.createTextNode('Hide Newsletter Sign-up'));
+        // icon for button
+        var iconI = doc.createElement('i');
+        iconI.className = 'icon-times';
+        iconI.setAttribute('aria-hidden', 'true');
+        // the actual button
+        var hideButton = doc.createElement('button');
+        hideButton.className ='only-icon newsletter-hide';
+        hideButton.setAttribute('type', 'button');
+        hideButton.appendChild(textSpan);
+        hideButton.appendChild(iconI);
+        // add to form
+        newsletterForm.appendChild(hideButton);
+        // add listener
+        hideButton.addEventListener('click', newsletterHandleHideClick, false);
+    }
+
+    function newsletterHide() {
+        // remove newsletter parent wiki-block
+        var wikiBlock = doc.querySelector('.wiki-block-newsletter');
+        wikiBlock.style.display = 'none';
+    }
+
+    function newsletterSaveHide() {
+        localStorage.setItem('newsletterHide', true);
+    }
+
+    function newsletterHandleHideClick() {
+        newsletterHide();
+        newsletterSaveHide();
```

Comment:
They're not always called together. 

When the newsletter is successfully submitted the localstorage parameter is set but the newsletter is not hidden (if it was hidden right away the user wouldn't see the success message).

When the user arrives on the page with the localstorage param set the form is hidden but the param is not saved. 

They're both called together when the user manually hides the newsletter.

I'd like to keep the functions separate.